kahuna/public/js/services/panel.js

```diff
@@ -51,8 +52,22 @@ panelService.factory('panelService', [function () {
         return newPanel(hidden, locked);
     }
 
+    // This is decoupled from the actual panels as we need the scope, which can be different in
+    // different contexts. You also might not want to have the state saved.
+    function setAndSaveState($scope, name, panel) {
+        const storeName = `${name}PanelState`;
```

Comment:
I've decided to just use a key for each so when we set, we don't have to read, update, write. It's just write.